test/asynchronous/test_srv_polling.py

```diff
@@ -186,6 +184,9 @@ def dns_resolver_response():
             ):
                 await assertion_method(expected_response, client)
 
+            # Close the client early to avoid affecting the next scenario run.
+            await client.close()
```

Comment:
Why does leaving the client open have side effects?

Reply:
We discussed in office hours: the thought is that the patching was affecting the previous client(s) that were not yet garbage collected.